assets/src/edit-story/app/history/useHistoryReducer.js

```diff
@@ -50,14 +57,30 @@ const reducer = (size) => (state, { type, payload }) => {
       // and clear `offset` and `replayState`.
       return {
         entries: [payload, ...state.entries.slice(state.offset)].slice(0, size),
+        globalHistoryLength: state.globalHistoryLength + 1,
+        hasHistoryChanged: true,
         offset: 0,
         replayState: null,
       };
 
     case REPLAY:
+      const historyChangedLengthOnUndo = state.globalHistoryLength - 1;
+      const historyChangedLengthOnRedo = state.globalHistoryLength + 1;
+      const newState = {
+        replayState: state.entries[payload],
+        globalHistoryLength:
+          payload < 0 ? historyChangedLengthOnRedo : historyChangedLengthOnUndo,
+      };
```

Comment:
Good point @barklund, but @spacedmonkey suggested https://github.com/google/web-stories-wp/pull/877#discussion_r402223355 to track undo changes (For instance, if the user undo until the initial state, the prevent window unload should be disabled, and if the user redo to some point ahead, it should be enabled again), which makes sense for me, or I misunderstand that? 

But from this point, what should I do? Do you have another way to propose track undo if not decreasing that value? I agree about rename to `versionNumber` or `historyID`. I liked both, but since it changes progressively, I would use versionNumber. If we decide to move forward tracking the undo/redo actions, I would prefer historyID.

Reply:
I agree. I think we just need one ever-incrementing value. It could be either truly global or it could be "number of changes since the last save". Either would solve the problem for us.